### PR TITLE
feat(middleware): Added some JWT middleware to decode from the bearer…

### PR DESCRIPTION
**Pull request by dbousamra**
… token and parse using the secret and function you pass in

### CODE REVIEW

File: src/main/scala/com/imageintelligence/http4c/middleware/JWTAuthMiddleware.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import org.http4s.headers.Authorization
+import org.http4s.server.AuthMiddleware
+
+import scalaz._
+import Scalaz._
+import scalaz.concurrent.Task
+
+object JWTAuthMiddleware {
+
+  def getBearerToken(req: Request): String \/ String = {
```
Comment (davidvuong) [inline src/main/scala/com/imageintelligence/http4c/middleware/JWTAuthMiddleware.scala]: Just curious, what's the point of making this a disjunction of String or String. Why not just String? Is it so you can use it in the `for` in `apply`?

**Comment by dbousamra** [comment]
The left side is the error message (if it fails). The right side is the
token (if it parses).

**Comment by davidvuong** [comment]
Sure but they're both just strings. Why not just returning a string rather than wrapping it up in something else?

File: src/main/scala/com/imageintelligence/http4c/middleware/JWTAuthMiddleware.scala (hunk 1 above)
Comment (AusCoder) [inline src/main/scala/com/imageintelligence/http4c/middleware/JWTAuthMiddleware.scala]: yes, but more fundamentally the left hand string represents some sort of failure and the right hand string is the token as a string. Once you have it in the Either monad, you can chain the operations and gather any errors together, as is done in the for comprehension in apply.

**Comment by dbousamra** [comment]
We need to represent the failure case. If we just returned a string, we have no way to figure out as a caller whether it parsed correctly or not. In most languages you would throw an exception instead of returning a value encompassing both errors. 

It's a bit like returning Option[String], except we want a verbose error message so we use Disjunction.

**Pull request merged**